src/librustc/ty/context.rs

```diff
@@ -1180,19 +1181,23 @@ macro_rules! direct_interners {
             }
         }
 
-        intern_method!($lt_tcx, $name: $method($ty, |x| x, |x| x, $needs_infer) -> $ty);)+
+        intern_method!($lt_tcx, $name: $method($ty, $alloc_method, |x| x, |x| x, $needs_infer) -> $ty);)+
     }
 }
 
 fn keep_local<'tcx, T: ty::TypeFoldable<'tcx>>(x: &T) -> bool {
     x.has_type_flags(ty::TypeFlags::KEEP_IN_LOCAL_TCX)
 }
 
+fn keep_local_slice<'tcx, T: ty::TypeFoldable<'tcx>>(xs: &[T]) -> bool {
```

Comment:
There's no point, it's a tiny helper. Besides, what you wrote doesn't work because the iterator is of references and for `keep_local` the type *behind* the reference is foldable.